graphql/schemabuilder/helpers.go

```diff
@@ -0,0 +1,392 @@
+	edgesSliceType := &graphql.NonNull{Type: &graphql.List{Type: edgeType}}
+
+	edgesSliceField := &graphql.Field{
+		Resolve: func(ctx context.Context, source, args interface{}, selectionSet *graphql.SelectionSet) (interface{}, error) {
+			if value, ok := source.(Connection); ok {
+				return value.Edges, nil
+			}
+			return nil, fmt.Errorf("error resolving edges in connection")
+		},
+		Type:           edgesSliceType,
+		ParseArguments: nilParseArguments,
+	}
+
+	fieldMap[makeGraphql("Edges")] = edgesSliceField
+
+	pageInfotype, err := sb.getType(reflect.TypeOf(PageInfo{}))
+	if err != nil {
+		return nil, err
+	}
+
+	pageInfoField := &graphql.Field{
+		Resolve: func(ctx context.Context, source, args interface{}, selectionSet *graphql.SelectionSet) (interface{}, error) {
+			if value, ok := source.(Connection); ok {
+				return value.PageInfo, nil
+			}
+			return nil, fmt.Errorf("error resolving pageinfo in connection")
+		},
+		Type:           pageInfotype,
+		ParseArguments: nilParseArguments,
+	}
+
+	fieldMap[makeGraphql("PageInfo")] = pageInfoField
+
+	retObject := &graphql.NonNull{Type: &graphql.Object{Name: "Connection", Description: "", Fields: fieldMap}}
+	return retObject, nil
+}
+
+func getConnection(res []interface{}, args ConnectionArgs) (interface{}, error) {
+
+	var nodes []Node
+	for _, val := range res {
+		if nodeVal, ok := val.(Node); ok {
+			nodes = append(nodes, nodeVal)
+		} else {
+			return nil, fmt.Errorf("node interface not implemented")
+		}
+	}
+	var edges []Edge
+	for _, val := range nodes {
+		edges = append(edges, Edge{Node: val, Cursor: val.GetCursor()})
+	}
+
+	if args.GetAfter() != nil {
+		edges = sliceAfter(edges, args.GetFirst(), *args.GetAfter())
+	} else if args.GetBefore() != nil {
+		edges = sliceBefore(edges, args.GetFirst(), *args.GetBefore())
+	} else {
+		return nil, fmt.Errorf("both after and before are nil")
+	}
+
+	return Connection{Edges: edges, PageInfo: PageInfo{HasNextPage: false}}, nil
+
+}
+
+func sliceAfter(edges []Edge, first int64, after string) []Edge {
+	var ret []Edge
+	i := 0
+	for ; i < len(edges) && edges[i].Cursor != after; i++ {
+	}
+	i++
+	if i == len(edges)+1 {
+		i = 0
+	}
+	for j := 0; j < int(first) && (i+j) < len(edges); j++ {
+		ret = append(ret, edges[i+j])
+	}
+	return ret
+}
+
+func sliceBefore(edges []Edge, first int64, before string) []Edge {
+	var ret []Edge
+	i := len(edges) - 1
+	for i = 0; i < len(edges) && edges[i].Cursor != before; i++ {
+	}
+	i--
+	if i >= len(edges) {
+		i = len(edges) - 1
+	}
+	for j := 0; j < int(first) && (i-j) >= 0; j++ {
+		ret = append(ret, edges[i-j])
+	}
+	return ret
+}
+
+func (o *Object) RegisterPaginatedField(name string, obj interface{}, f interface{}) {
+	o.PaginatedFields = append(o.PaginatedFields,
+		PaginationObject{
+			Name:     name,
+			NodeType: reflect.TypeOf(obj),
+			Fun:      f,
+		})
+}
+
+func (sb *schemaBuilder) GetPaginatedField(typ reflect.Type, f interface{}) (*graphql.Field, error) {
+	fun := reflect.ValueOf(f)
```

Comment:
could you comment (in the PR) about how we can refactor this? how much is identical for `buildFunction`? how much is different?